src/drivers/cmakeFileApiDriver.ts

```diff
@@ -291,7 +291,7 @@ export class CMakeFileApiDriver extends CMakeDriver {
             log.trace(result.stderr);
             log.trace(result.stdout);
             if (result.retc === 0) {
-                if (!configurePreset) {
+                if (!configurePreset || (configurePreset && defaultConfigurePresetName && configurePreset.name === defaultConfigurePresetName)) {
```

Comment:
Just curious based on the `defaultConfigurePresetName && configurePreset.name === defaultConfigurePresetName` check, is there a known case where `configurePreset` is truthy but `configurePreset.name` is falsy?

Reply:
I don't believe so. My understanding is that there MUST be a name for a CMake Preset, because the name should be the unique identifier of it compared to other presets. If we were using `displayName` then I think there could be a case, but not for `name`. Thanks!

Reply:
I don't remember why there would need to be a condition around setting `this._needsReconfigure`. It used to be set unconditionally here until #2636.

Reply:
And if you make a change to the condition here, I would think you'd want to do it for the other drivers as well.

Reply:
@bobbrow I had looked through that PR as well while implementing this. The reason it's not set unconditionally is basically that it's using that variable primarily for extension state, thus the commands, and auto-configure, etc. However, tasks can be configuring for any preset, which might not match what is chosen in the UI. 

I'll check about changing the condition for other drivers.